lease.go

```diff
@@ -19,30 +19,27 @@ type WgRecord struct {
 	expires time.Time
 }
 
-var (
-	// wgRecords holds a PublicKey -> IP address and lease time mapping
-	WgRecords map[string]*WgRecord
+type FileLeaseManager struct {
+	wgRecords map[string]*WgRecord
 	filename  string
-	CIDR      *net.IPNet
-	LeaseTime time.Duration
-)
+	cidr      *net.IPNet
+	leaseTime time.Duration
+	ip        net.IP
+}
 
-func initWithFile(f string, cidr *net.IPNet, leasetime time.Duration) error {
```

Comment:
I think this should be `NewFileLeaseManager(filename string, ....) (FileLeaseManager, error)`